src/jimgw/single_event/detector.py

```diff
@@ -40,32 +40,44 @@ class Detector(ABC):
     data: jd.Data
     psd: jd.PowerSpectrum
 
-    frequency_bounds: tuple[float, float] = (0., float("inf"))
+    frequency_bounds: tuple[float, float] = (0.0, float("inf"))
+
+    _sliced_frequencies: Float[Array, " n_sample"] = jnp.array([])
+    _fd_data_slice: Float[Array, " n_sample"] = jnp.array([])
+    _psd_slice: Float[Array, " n_sample"] = jnp.array([])
+
+    @property
+    def epoch(self):
+        """The epoch of the data."""
+        return self.data.epoch
 
     @abstractmethod
     def fd_response(
         self,
         frequency: Float[Array, " n_sample"],
         h_sky: dict[str, Float[Array, " n_sample"]],
         params: dict,
+        trigger_time: Float = 0.0,
         **kwargs,
-    ) -> Float[Array, " n_sample"]:
+    ) -> Complex[Array, " n_sample"]:
         """Modulate the waveform in the sky frame by the detector response in the frequency domain.
-        
+
         Args:
             frequency (Float[Array, " n_sample"]): Array of frequency samples.
-            h_sky (dict[str, Float[Array, " n_sample"]]): Dictionary mapping polarization names 
-                to frequency-domain waveforms. The keys are polarization names (e.g., 'plus', 'cross') 
+            h_sky (dict[str, Float[Array, " n_sample"]]): Dictionary mapping polarization names
+                to frequency-domain waveforms. The keys are polarization names (e.g., 'plus', 'cross')
                 and values are complex strain arrays.
             params (dict): Dictionary of source parameters including:
                 - ra (float): Right ascension in radians
                 - dec (float): Declination in radians
                 - psi (float): Polarization angle in radians
                 - gmst (float): Greenwich mean sidereal time in radians
+                - t_c (Float): Difference between geocent time and trigger time in sec
+            trigger_time (Float): Trigger time of the data in seconds.
```

Comment:
I think the trigger time should be a params in the dict instead of being its own parameters

Reply:
If gmst is no longer used, one can replace gmst with trigger_time